src/main/kotlin/blackjack/model/participant/HandCards.kt

```diff
@@ -0,0 +1,25 @@
+package blackjack.model.participant
+
+import blackjack.model.card.Card
+import blackjack.model.card.Denomination
+
+class HandCards {
+    private val _cards = mutableListOf<Card>()
+    val cards: List<Card>
+        get() = _cards.toList()
+
+    fun add(cards: List<Card>) {
+        _cards.addAll(cards)
+    }
+
+    fun getCardSum(burstCondition: Int): Int {
+        var cardSum = _cards.sumOf { it.getScore() }
+        cardSum =
+            _cards.fold(cardSum) { total, card ->
+                if (!card.isAce()) return@fold total
+                if (total + Denomination.ADDITIONAL_ACE_SCORE > burstCondition) return total
+                total + Denomination.ADDITIONAL_ACE_SCORE
+            }
+        return cardSum
+    }
```

Comment:
```suggestion
    fun getCardSum(burstCondition: Int): Int {
        return _cards.fold(
            _cards.sumOf { it.getScore() }
        ) { total, card ->
            if (!card.isAce()) return@fold total
            if (total + Denomination.ADDITIONAL_ACE_SCORE > burstCondition) return total
            total + Denomination.ADDITIONAL_ACE_SCORE
        }
    }
```

이렇게 하면 var 을 없애고 불변으로 나타낼 수 있지 않을까 해서 제안해봅니다.
개인적으로 fold 의 operation 람다에 if 문이 너무 많이 연속되었다는 생각이 드는데,
if 문은 연속하지 않도록 쪼개는 리팩토링이 있으면 좋을 것 같습니다.
참고로 지나치게 엄격할 수도 있는 기준이므로 꼭 반영해주시지 않아도 괜찮습니다.

Reply:
말씀하신 방법대로 구현하면 cardSum이라는 별도의 가변 변수를 생성하지 않아도 되어서 훨씬 깔끔해지네요!! 감사합니다 코드에 적용해보겠습니다. ㅎㅎ

사실 저도 fold 내의 if문이 마음에 안 들어서 수정하려고 여러번 시도해봤는데요... 첫 번째 if는 fold만 return하고 두 번째 if는 함수 자체를 return해서 어떻게 리팩토링해야 할지 잘 모르겠습니다. 일단 더 고민해보겠습니다!!